File: javascript/src/models/sinkStatusIn.ts
// this file is @generated

export enum SinkStatusIn {
  Enabled = "enabled",
  Disabled = "disabled",
}

export const SinkStatusInSerializer = {
  _fromJsonObject(object: any): SinkStatusIn {
    return object;
  },

  _toJsonObject(self: SinkStatusIn): any {
    return self;
  },
};

File: kotlin/lib/src/main/kotlin/models/SinkStatusIn.kt
// This file is @generated
package com.svix.kotlin.models

import com.svix.kotlin.ToQueryParam
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class SinkStatusIn : ToQueryParam {
    @SerialName("enabled") ENABLED,
    @SerialName("disabled") DISABLED;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
